app/ante/sigverify.go

```diff
@@ -29,16 +30,26 @@ import (
 // Failure in RecheckTx will prevent tx to be included into block, especially when CheckTx succeed, in which case user
 // won't see the error message.
 func VerifyEthSig(tx sdk.Tx, signer ethtypes.Signer) error {
+	errChan := make(chan error, len(tx.GetMsgs()))
 	for _, msg := range tx.GetMsgs() {
```

Comment:
@yihuang could reduce from 2s to 0.2s for 3k msgs when parallel

Reply:
Only for large batch tx?

Reply:
Can you do this separately, and avoid extra overhead for normal cases where tx has single msg.

Reply:
should be same for small batch, but I only filter out when CheckTx > [1s](https://github.com/mmsqe/cosmos-sdk/commit/5472e9cf6643340c1f915eb37369583f3111a634#diff-389d02f089ab55575400f5300d3c5cfe277fde5f04bd0fe6a85456a7252d1099L357)

Reply:
I mean for tx that has a single msg, this change is purely overhead, no benifit.